### PR TITLE
Remove Snappy

### PR DESCRIPTION
**Pull request by obelisk**
We only had snappy to link RocksDB with it but now that we have ZSTD, we don't need to pull it in anymore.

### CODE REVIEW

**Comment by obelisk** [comment]
Also investigating if we can remove LZ4 as well

**Comment by osqueryer** [comment]
:-1: The commit ef833989f2a7fc4df3a3e33ffcdda9eb37b5bb8e (Job results: [1581](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1581/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit ef833989f2a7fc4df3a3e33ffcdda9eb37b5bb8e (Job results: [5479](https://jenkins.osquery.io/job/osqueryPullRequestLinux/5479/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit ef833989f2a7fc4df3a3e33ffcdda9eb37b5bb8e (Job results: [2372](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2372/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit ef833989f2a7fc4df3a3e33ffcdda9eb37b5bb8e (Job results: [2373](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2373/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit ef833989f2a7fc4df3a3e33ffcdda9eb37b5bb8e (Job results: [1582](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1582/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit ef833989f2a7fc4df3a3e33ffcdda9eb37b5bb8e (Job results: [5480](https://jenkins.osquery.io/job/osqueryPullRequestLinux/5480/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 8f136cbf968fdefb07f15e5d847277f3a4a43363 (Job results: [1583](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1583/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit 8f136cbf968fdefb07f15e5d847277f3a4a43363 (Job results: [5481](https://jenkins.osquery.io/job/osqueryPullRequestLinux/5481/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit 8f136cbf968fdefb07f15e5d847277f3a4a43363 (Job results: [1584](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1584/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit 8f136cbf968fdefb07f15e5d847277f3a4a43363 (Job results: [5482](https://jenkins.osquery.io/job/osqueryPullRequestLinux/5482/)) failed one or more tests (**Linux**).

**Comment by theopolis** [comment]
You'll most likely need to uninstall `snappy` for this to work. I can take care of this on the build hosts, so just mark this in-progress for now.

File: tools/provision.sh [hunk 1]
```diff
@@ -130,14 +130,12 @@ function platform_darwin_main() {
...
 
   # List of LLVM-compiled dependencies.
-  brew_dependency osquery/osquery-local/lz4
   brew_dependency osquery/osquery-local/libmagic
   brew_dependency osquery/osquery-local/pcre
   brew_dependency osquery/osquery-local/boost
   brew_dependency osquery/osquery-local/asio
   brew_dependency osquery/osquery-local/cpp-netlib
   brew_dependency osquery/osquery-local/google-benchmark
-  brew_dependency osquery/osquery-local/snappy
```
Comment (theopolis) [inline tools/provision.sh]: There should be a `brew_uninstall` bash function, use that, then bump the `rocksdb` revision again to uninstall from build hosts.

**Comment by theopolis** [comment]
I'm not sure about Windows, you may need to rebuild the `rocksdb` package, since it is most likely statically compiling snappy too.

**Pull request merged**